src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -69,10 +71,9 @@ private Task PackageInitializationBackgroundThreadAsync(PackageLoadTasks package
     {
         try
         {
-            this.RegisterService<ICSharpTempPECompilerService>(async ct =>
+            ((IServiceContainer)this).AddService(typeof(ICSharpTempPECompilerService), (_, _) =>
             {
                 var workspace = this.ComponentModel.GetService<VisualStudioWorkspace>();
-                await JoinableTaskFactory.SwitchToMainThreadAsync(ct);
                 return new TempPECompilerService(workspace.Services.GetService<IMetadataService>());
             });
         }
```

Comment:
@jasonmalinowski i can also take this further.  If we're just registering a callback for some future point, then no need to have this as a task on PackageInitializationBackgroundThreadAsync.  We can just early register this.  

Note: i don't know this part of the system super well.  so LMK if that's undesirable for some reason.

Reply:
I'd say let's just leave that as is for now; that way everything is going through the registration system we have rather than a mix of things.

Ideally, I'd like to be to a point where we _can_ delete all the fancy registration stuff, not because it's bad, but hopefully we're to the point where we have so little in our package startup that we don't really need the fanciness again. But we're not remotely there yet.